### DISCUSSION TITLE
Setup help

### DISCUSSION THREAD

**Discussion by nlindblo** [shannonhochkins/ha-component-kit#62] [General]
Hello,

I am new to react/node and front end development in general and I am struggling to progress via the setup instructions. I have HA running in a Kubernetes container and would like to understand how I can put the code on a separate machine (like my laptop), configure the connection to HA for the API calls and do the development from there, rather than using the deploy script that copies the code to the HA instance (I can figure that out later once I developed my dashboard). 

Apologies if this is obvious, but I would appreciate a little push just to get going as the this looks like the best dashboard I seen so far.

Thanks

**Comment by shannonhochkins** [comment]
Hey mate, I'm not sure where you're actually stuck, you mentioned you haven't managed to get past the setup instructions, what exactly has gone wrong? These packages automatically ask you to login to home assistant so there's no need to communicate with the rest api, currently this project is purely websocket information for dashboard rendering

However this will be something I'll be integrating down the track for these packages as the rest api can obviously do more & retrive than the websockets can

I'm still not quite sure what the problem is, after installed and you've run the npm script to run on a dev server, did it ask you to login to home assistant?

If you haven't gotten that far I'm happy to help you :)

**Reply by nlindblo** [reply]
Sorry, as soon as I posted I saw the instruction in the command line for how to start the application, so I got the basic install up and running now on my Mac, so all good for the time being.

Thanks for the quick response anyway

**Reply by shannonhochkins** [reply]
That's good! Happy to help, I suggest watching a few YouTube videos on basic react, it should be all you'll need to get something going!

If you used the `npm create` command it should give you a bunch of tools to help you when you may go wrong :)

**Comment by shannonhochkins** [comment]
I will close this as there's not been any further comments. Feel free to open it back up if you need help!
